pkg/util/winutil/service_test.go

```diff
@@ -511,6 +511,76 @@ func (s *StartStopServiceRaceConditionTestSuite) TestStopMainWhenDependentServic
 	assert.Equal(s.T(), status.State, svc.Stopped, "`Dep2` service should be stopped")
 }
 
+// Test start a service after it has been just started. This is to test
+// that starting a service when it is in START_PENDING state is handled correctly
+func (s *StartStopServiceRaceConditionTestSuite) TestStartServiceAfterStartService() {
+	// SETUP: Start main service only
+	err := s.mainSvc.Start()
+	assert.Nil(s.T(), err, "Failed to start `main` service: %v", err)
+
+	// Immediately start it again, since main service most likely will be in
+	// the start pending state at the moment of the second start. It should start
+	// successfully without any error
+	err = StartService(s.mainSvcName)
+	assert.Nil(s.T(), err, "`Main` service should start successfully: %v", err)
+}
+
+// Test start a service after it has been just stopped. This is to test
+// that starting a service when it is in STOP_PENDING state is handled correctly
+func (s *StartStopServiceRaceConditionTestSuite) TestStartServiceAfterStopService() {
+	// SETUP: Start, wait for starting and stop externally main service
+	// to make sure it is in the stop pending state
+	err := s.mainSvc.Start()
+	assert.Nil(s.T(), err, "Failed to start `main` service: %v", err)
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	err = WaitForState(ctx, s.mainSvcName, svc.Running)
+	assert.Nil(s.T(), err, "`Main` service should be running: %v", err)
+	_, err = s.mainSvc.Control(svc.Stop)
+	assert.Nil(s.T(), err, "`Main` service should be stopping: %v", err)
+
+	// Immediately start it again, since main service most likely will be in
+	// the stop pending state at the moment of the second start. It should start
+	// successfully without any error
+	err = StartService(s.mainSvcName)
+	assert.Nil(s.T(), err, "`Main` service should start successfully: %v", err)
+}
+
+// Test stop a service after it has been just started. This is to test
+// that stoppiong a service when it is in START_PENDING state is handled correctly
+func (s *StartStopServiceRaceConditionTestSuite) TestStopServiceAfterStartService() {
```

Comment:
I think I debugged it to make sure that it is in the pending state by the time it was attempted to stop. The problem is actually, as I have tried to state that in some cases being in the pending state is Ok for a service to be started/stopped. The command is accepted. But for the same service I found a hard way, and this is the reason for change that in some cases it would fail (and state of the service has a hint about it about accepting cmd).

But in general I agree. It did cross my mind to create own service for finetuning unit tests. However I did not want to include binaries into the repo. I did think of ability to add build during unit test on the fly but it was bigger effort than I was willing to commit to.